sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -108,16 +114,17 @@ case class AdaptiveSparkPlanExec(
     // around this case.
     EnsureRequirements(optimizeOutRepartition = requiredDistribution.isDefined),
```

Comment:
is it safer to pass in the `requiredDistribution` to this `EnsureRequirements` instance as well?

Reply:
combined, ensure we run the same `EnsureRequirements` twice